### ISSUE TITLE
Wrong text for viewing profile in Who's online

### ISSUE THREAD

**Issue by frandominguezl** [SimpleMachines/SMF#7004]
### Steps to reproduce
1. Navigate to Who's online when someone is viewing a profile

### Additional information/references
![image](https://user-images.githubusercontent.com/10365222/130350663-93fa5978-a457-45d6-b9e5-23edc93a3eaf.png)

**Comment by wintstar** [comment]
I can not confirm

![smf_online_bug](https://user-images.githubusercontent.com/2753863/130352515-41f15896-c660-40c7-9c08-2def85a1b543.png)

**Comment by frandominguezl** [comment]
Then perhaps this is a SMF site issue? @jdarwood007

**Comment by Sesquipedalian** [comment]
I also cannot reproduce this on a test forum. 

I do see it happening on simplemachines.org, but only in some cases:
![Screen Shot 2021-08-22 at 2 03 49 PM](https://user-images.githubusercontent.com/191731/130368517-8fcd38dc-4fa3-4675-bb3f-dca15cfc65ac.png)

**Comment by wintstar** [comment]
I can remember that I once had this under 2.0.15. As far as I can remember it was a problem with the permissions.

**Comment by Sesquipedalian** [comment]
I don't have access to the database on simplemachines.org, but I think I found some issues in the code that could cause this.

The immediate cause appears to be that we aren't bailing out properly when we encounter malformed JSON in the log_online table. In SMF 2.0.x, we stored the URL that a user was currently viewing as a serialized string, but in SMF 2.1 we store it as a JSON string. When `unserialize()` encounters a malformed string, it returns `false`. In contrast, when `smf_json_decode()` encounters a malformed string, it returns `array()`. But Who.php still expects `false` if there was an error, and so we don't bail out when we should:
https://github.com/SimpleMachines/SMF2.1/blob/4bbda49cf841a5008539a72b75015dce610e31d4/Sources/Who.php#L158-L160
https://github.com/SimpleMachines/SMF2.1/blob/4bbda49cf841a5008539a72b75015dce610e31d4/Sources/Who.php#L331-L333

Behind that immediate cause lies the second issue, which is that we are writing malformed JSON to the log at all. This is an issue that has come up in the past, and we thought we'd solved it (see #4049 & #4070, and #5653 & #5680), but apparently we have not entirely.
